### PR TITLE
[CAMEL-15872] JWT Authorization Type Configuration Option to GraphQL Component

### PR DESCRIPTION
**Pull request by tapiiron**
Addresses wish CAMEL-15872

### CODE REVIEW

File: components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java [hunk 1]
```diff
@@ -187,6 +193,18 @@ public String getQuery() {
         return query;
     }
 
+    /**
+     * The JWT Authorization type. Default is Bearer.
+     * @param jwtAuthorizationType
```
Comment (oscerd) [inline components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java:198]: Please remove param annotation.

File: components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java [hunk 2]
```diff
@@ -110,8 +112,12 @@ private CloseableHttpClient createHttpClient() {
             httpClientBuilder.setProxy(new HttpHost(hostname, port));
         }
         if (accessToken != null) {
+            String authType = "Bearer";
+            if(this.jwtAuthorizationType != null) {
```
Comment (oscerd) [inline components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java:116]: If you run a full build, this should be formatted to respect the camel code style standards.

**Review by oscerd — CHANGES_REQUESTED** [review]
Just two minor findings.

File: components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java (hunk 1 above)
Comment (tapiiron) [reply components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java]: Removed

**Review by oscerd — APPROVED** [review]

File: components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java (hunk 2 above)
Comment (tapiiron) [reply components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java]: That missed my eye but should be fixed now

File: components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java [hunk 3]
```diff
@@ -65,6 +65,8 @@
     @UriParam(label = "security", secret = true)
     private String password;
     @UriParam
+    private String jwtAuthorizationType;
```
Comment (davsclaus) [inline components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java:68]: Can you set label = "security", and default = "Bearer" in the uri param annotation

Comment (tapiiron) [reply components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java:68]: Sure thing, added those now

**Review by davsclaus — APPROVED** [review]

**Pull request merged**